### PR TITLE
Cache capabilities

### LINKED ISSUE #9854: Outsource ThemeUtils
Currently we any ThemeUtils needs to access capabilities and thus database, which sometimes results in >5 queries per activity.

I can imagine that a global state, provided via dagger (with `@singleton`) will reduce this and might enhance performance.

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix #9854 

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
@AlvaroBrey we also have SessionMixin, which has capabilities bound to the session.
It now feels that we do it twice…

But it helps:
adb logcat -d | grep getCapability | wc is down from 316 to 18 🎉
(counts are not calls, as logcat is very verbos)

**Review by AlvaroBrey — COMMENTED** [review]
Looks good, but it would be better to make CapabilityUtils and ThemeColorUtils injectable singletons, for better testability, rather than use static methods.

Additionally, I see there's still a lot of usages of `FileDataStorageManager.getCapability(String accountName)` throughout the codebase. If the intention is not to address those in this same PR, we should probably mark it as `@Deprecated` with a comment to use `CapabilityUtils` instead.

**Comment by AlvaroBrey** [comment]
> we also have SessionMixin, which has capabilities bound to the session.
> It now feels that we do it twice…

We can change `SessionMixin` so that it calls `CapabilityUtils` instead of storing the capabilities itself:

```
    val capabilities: OCCapability?
        get() = getUser()
            .map { CapabilityUtils.getCapability(it, activity) }
            .orElse(null)
```

**Comment by AlvaroBrey** [comment]
@tobiasKaminsky is this ready for review?

File: app/src/main/java/com/owncloud/android/utils/theme/CapabilityUtils.java [hunk 1]
```diff
@@ -41,10 +48,43 @@ public static OCCapability getCapability(Context context) {
...
         if (user != null) {
-            FileDataStorageManager storageManager = new FileDataStorageManager(user, context.getContentResolver());
-            return storageManager.getCapability(user.getAccountName());
+            return getCapability(user, context);
         } else {
             return new OCCapability();
         }
     }
+
+    public static OCCapability getCapability(Account acc, Context context) {
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/utils/theme/CapabilityUtils.java:57]:
```suggestion
    /**
     * @deprecated use {@link #getCapability(User, Context)} instead
     */
    @Deprecated
    public static OCCapability getCapability(Account acc, Context context) {
```

**Review by AlvaroBrey — APPROVED** [review]
Looks good! Minor comment about deprecation

**Pull request merged**